cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -1343,6 +1349,32 @@ fromBabbagePParamsUpdate
                                             strictMaybeToMaybe _coinsPerUTxOByte
     }
 
+-- | Bundle cardano-api representation and ledger representation of protocol paramets together so
+-- they can be computed once and passed around rather than re-computed unecessarily.
+--
+-- The consructor arguments are intentionally lazy so that the values are not computed if not used
+-- (which may be the case for some code paths).
+data BundledProtocolParameters era where
+  BundleAsByronProtocolParameters
+    :: ProtocolParameters
+    -> BundledProtocolParameters ByronEra
+  BundleAsShelleyBasedProtocolParameters
+    :: ShelleyBasedEra era
+    -> ProtocolParameters
+    -> Ledger.PParams (ShelleyLedgerEra era)
+    -> BundledProtocolParameters era
+
+bundleProtocolParams :: CardanoEraStyle era -> ProtocolParameters -> BundledProtocolParameters era
```

Comment:
This would required moving the call site of `bundleProtocolParams` [here](https://github.com/input-output-hk/cardano-node/blob/newhoggy/reduce-number-of-calls-to-toLedgerPParams/cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs/#L1240), to before the `case` match.

Is that okay?

Reply:
Done!